### PR TITLE
Fixing HDF5 app compilation for windows

### PR DESCRIPTION
**Pull request by roigcarlo**
**📝 Description**
Mostly symbol exports, but also a odd bug in the mpi communicator.

Needs -DHDF5_ROOT to point at the compilation / installation dir.

**🆕 Changelog**
- Hdf5 application can now be compiled in win.

### CODE REVIEW

File: applications/HDF5Application/custom_io/hdf5_container_gauss_point_output.cpp [hunk 1]
```diff
@@ -315,7 +315,7 @@ void SetDataBuffer(Variable<TDataType> const& rVariable,
     }
 
     NumberOfGaussPoints = rDataCommunicator.MaxAll(NumberOfGaussPoints);
-    flat_data_size = rDataCommunicator.MaxAll(flat_data_size);
+    flat_data_size = rDataCommunicator.MaxAll((int)flat_data_size);
```
Comment (roigcarlo) [inline applications/HDF5Application/custom_io/hdf5_container_gauss_point_output.cpp:318]: This is the odd fix. I am not sure why it is complaining, but it cannot deduce the type automatically from std::size_t

**Review by sunethwarna — APPROVED** [review]
Nice addition. Thanks @roigcarlo . Hmm... The CI failure is not related this ryt?

**Comment by roigcarlo** [comment]
Nope, that's from another PR. @RiccardoRossi I think this is from the Particle utilities. The first comparison is always true if the type is std::size_t

**Review by RiccardoRossi — APPROVED** [review]
looks ok to me

**Comment by RiccardoRossi** [comment]
the failure for the intel compiler is addressed in #10368

**Pull request merged**